source/client/process_impl.cc

```diff
@@ -153,6 +191,35 @@ ProcessImpl::ProcessImpl(const Options& options, Envoy::Event::TimeSystem& time_
   configureComponentLogLevels(spdlog::level::from_str(lower));
 }
 
+absl::StatusOr<ProcessPtr>
+ProcessImpl::CreateProcessImpl(const Options& options, Envoy::Event::TimeSystem& time_system,
+                               const std::shared_ptr<Envoy::ProcessWide>& process_wide) {
+  std::unique_ptr<ProcessImpl> process(new ProcessImpl(options, time_system, process_wide));
+
+  absl::StatusOr<Bootstrap> bootstrap = createBootstrapConfiguration(
+      *process->dispatcher_, process->options_, process->number_of_workers_);
+  if (!bootstrap.ok()) {
+    ENVOY_LOG(error, "Failed to create bootstrap configuration: {}", bootstrap.status().message());
+    process->shutdown();
+    return bootstrap.status();
+  }
+
+  // Ideally we would create the bootstrap first and then pass it to the
+  // constructor of Envoy::Api::Api. That cannot be done because of a circular
+  // dependency:
+  // 1) The constructor of Envoy::Api::Api requires an instance of Bootstrap.
+  // 2) The bootstrap generator requires an Envoy::Event::Dispatcher to resolve
+  //    URIs to IPs required in the Bootstrap.
+  // 3) The constructor of Envoy::Event::Dispatcher requires Envoy::Api::Api.
```

Comment:
Can we create an issue to potentially remove this circular dependency?  I wonder if one of these things (dispatcher?) doesn't really need what it current requires.

Reply:
I was on the fence of making a similar comment earlier... +1 from me.

Reply:
Opened https://github.com/envoyproxy/nighthawk/issues/735.